syntax = "proto3";

package oaa.proto.messages;

// Media playback metadata (msgId 0x8003 on Media channel)
// Sent by phone when track changes â€” includes album art
message MediaPlaybackMetadata {
    optional string title = 1;
    optional string artist = 2;
    optional string album = 3;
    // Album art as PNG image data
    optional bytes album_art = 4;

    // Fields 5-6 discovered via APK proto decoder (nmi.java, AA v16.1)
    optional bool is_playing = 5;       // Whether media is actively playing
    optional string album_art_url = 6;  // URL for album art (alternative to embedded bytes)
}
